mailbox/api/src/main/java/org/apache/james/mailbox/model/SimpleMailboxAnnotation.java

```diff
@@ -0,0 +1,328 @@
+        }
+    }
+
+    /**
+     * Default implementation of {@link MailboxACLEntryKey}.
+     */
+    public static class SimpleMailboxAnnotationEntryKey implements MailboxAnnotationEntryKey {
+        private final String name;
+
+        /**
+         * Creates a new instance of SimpleMailboxAnnotationEntryKey from the given
+         * serialized {@link String}.
+         * 
+         * @param serialized
+         */
+        public SimpleMailboxAnnotationEntryKey(String serialized) {
+
+            if (serialized == null) {
+                throw new IllegalStateException("Cannot parse null to a " + getClass().getName());
+            }
+            serialized = serialized.trim();
+            if (serialized.length() == 0) {
+                throw new IllegalStateException("Cannot parse an empty string to a " + getClass().getName());
+            }
+            if (serialized.charAt(0) != SLASH_CHARACTER) {
+                throw new IllegalStateException("The annotation does not start with a slash " + getClass().getName());
+            }
+            if (serialized.contains("" + ASTERISK_CHARACTER)) {
+                throw new IllegalStateException("The annotation contains at least a '*' character " + getClass().getName());
+            }
+            if (serialized.contains("" + PERCENT_CHARACTER)) {
+                throw new IllegalStateException("The annotation contains at least a '%' character " + getClass().getName());
+            }
+            if (serialized.contains("" + SLASH_CHARACTER + SLASH_CHARACTER)) {
+                throw new IllegalStateException("The annotation contains 2 consecutive '/' character " + getClass().getName());
+            }
+            if (serialized.endsWith("" + SLASH_CHARACTER)) {
+                throw new IllegalStateException("The annotation ends with '/' character " + getClass().getName());
+            }
+            if (NAME_ANNOTATION_PATTERN.matcher(serialized).find()) {
+                throw new IllegalStateException("The annotation contains non-ASCII character except some special characters " + getClass().getName());
+            }
+            this.name = serialized;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (o instanceof MailboxAnnotationEntryKey) {
+                MailboxAnnotationEntryKey other = (MailboxAnnotationEntryKey) o;
+                return this.name.equals(other.getName());
+            } else {
+                return false;
+            }
+        }
+
+        /*
+         * (non-Javadoc)
+         * 
+         * @see org.apache.james.mailbox.MailboxACL.MailboxACLEntryKey#getName()
+         */
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public int hashCode() {
+            return name.hashCode();
+        }
+
+        @Override
+        public String toString() {
+            return new StringBuilder(name.length()).append(name).toString();
```

Comment:
and to be honest, StringBuilder is almost useless, the compiler call it for you when you do string concat